agent/command/results_xunit_parser.go

```diff
@@ -106,6 +108,14 @@ func (tc testCase) toModelTestResultAndLog(conf *internal.TaskConfig) (testresul
 	// Replace spaces, dashes, etc. with underscores.
 	res.TestName = util.CleanForPath(res.TestName)
 
+	if math.IsNaN(float64(tc.Time)) {
+		logger.Task().Errorf("Test '%s' time was NaN, its calculated duration will be incorrect", res.TestName)
+	}
+	if math.IsInf(float64(tc.Time), 0) || math.IsInf(float64(tc.Time), -1) {
```

Comment:
nit: it wasn't obvious to me why we passed in 0 to IsInf so could we add a little comment saying its to check for both -inf and inf